packages/zowe-explorer/src/Profiles.ts

```diff
@@ -1327,26 +1331,20 @@ export class Profiles extends ProfilesCache {
 
     private async updateBaseProfileFileLogin(profile: IProfileLoaded, updProfile: IProfile) {
         const upd = { profileName: profile.name, profileType: profile.type };
-        // const config = (await this.getProfileInfo()).getTeamConfig();
-        const config = ImperativeConfig.instance.config;
-        const { user, global } = config.api.layers.find(upd.profileName);
-        const profilePath = config.api.profiles.expandPath(upd.profileName);
-        config.api.layers.activate(user, global);
-        config.set(`${profilePath}.properties.tokenType`, updProfile.tokenType);
-        config.set(`${profilePath}.properties.tokenValue`, updProfile.tokenValue);
-        await config.save();
+        const mProfileInfo = await this.getProfileInfo();
+        const setSecure = mProfileInfo.isSecured();
+        await mProfileInfo.updateProperty({ ...upd, property: "tokenType", value: updProfile.tokenType });
+        await mProfileInfo.updateProperty({ ...upd, property: "tokenValue", value: updProfile.tokenValue, setSecure });
     }
 
     private async updateBaseProfileFileLogout(profile: IProfileLoaded) {
         const upd = { profileName: profile.name, profileType: profile.type };
-        // const config = (await this.getProfileInfo()).getTeamConfig();
-        const config = ImperativeConfig.instance.config;
-        const { user, global } = config.api.layers.find(upd.profileName);
-        const profilePath = config.api.profiles.expandPath(upd.profileName);
-        config.api.layers.activate(user, global);
-        config.set(`${profilePath}.properties.tokenType`, undefined);
-        config.set(`${profilePath}.properties.tokenValue`, undefined);
-        await config.save();
+        const mProfileInfo = await this.getProfileInfo();
+        const setSecure = mProfileInfo.isSecured();
+        const prof = mProfileInfo.getAllProfiles(profile.type).find((prof) => prof.profName === profile.name);
```

Comment:
this is the only time I see `prof` in the changed code. in the const and as the find variable

Reply:
thanks! 😋 
Already push a quick fix for this!